mkt/extensions/validation.py

```diff
@@ -0,0 +1,92 @@
+import json
+from zipfile import BadZipfile, ZipFile
+
+from rest_framework.exceptions import ParseError
+from tower import ugettext as _
+
+
+class ExtensionValidator(object):
+    """
+    Firefox OS Add-on validator. If validation fails, will raise an instance of
+    rest_framework.exceptions.ParseError containing information about the
+    error.
+    """
+    errors = {
+        'BAD_CONTENT_TYPE': _('The file sent has an unsupported content-type'),
+        'INVALID_JSON': _(("'manifest.json' in the archive is not a valid JSON"
+                           " file.")),
+        'INVALID_ZIP': _('The file sent is not a valid ZIP file.'),
+        'NAME_MISSING': _('There is no `name` property in the manifest.'),
+        'NAME_NOT_STRING': _('The `name` property must be a string.'),
+        'NAME_TOO_LONG': _(('The `name` property cannot be longer than 45 '
+                            'characters.')),
+        'NAME_TOO_SHORT': _(('The `name` property must be at least 1 character'
+                             ' long.')),
+        'NO_MANIFEST': _(("The archive does not contain a 'manifest.json' "
+                          "file.")),
+    }
+    valid_content_types = (
+        'application/octet-stream',
+        'application/zip',
+    )
+
+    def __init__(self, file_obj=None):
+        """
+        Validate the uploaded file:
+
+        * Ensure that it is a valid zip file.
+        * Ensure that it contains a valid manifest.json file.
+        * Validate the manifest against the spec.
+        """
+        self.file = file_obj
+        self.manifest_file = None
+        self.manifest_json = None
```

Comment:
Could probably run `self.validate` here, but this makes it eminently more testable.